servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusDatabaseAdapterConfig.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.config;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithConverter;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
+import org.projectnessie.versioned.persist.nontx.NonTransactionalDatabaseAdapterConfig;
+
+@ConfigMapping(prefix = "nessie.adapter")
+@RegisterForReflection(targets = TrimmedStringConverter.class)
+public interface QuarkusDatabaseAdapterConfig extends NonTransactionalDatabaseAdapterConfig {
+
+  @WithName("key-prefix")
+  @WithDefault(DEFAULT_KEY_PREFIX)
+  @WithConverter(TrimmedStringConverter.class)
+  @Override
+  String getKeyPrefix();
+
+  @WithName("parents-per-global-commit")
```

Comment:
Are we planning to add a comment for every config explaining the purpose of the particular config?

Reply:
We have javadoc on base methods (in superclasses)

Reply:
I see, nevertheless. isn't easier to have some docs here that explains the purpose of the config class when you read the config interface? Also, IMHO it will make our life easier when we decide to automate generating the config docs through Quarkus `@ConfigMapping` annotation.

Reply:
I'd expect Quarkus build to be able to extract javadoc from superclasses.

I'll add a top-level javadoc to this class too.

Reply:
updated in e094dc26344a0f68b3d7c3adba03d0d7f118280d